# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1521703169 18000
#      Thu Mar 22 02:19:29 2018 -0500
# Branch stable
# Node ID 2eca27a6c87689b2de818c52eaff7d59d15aaf1b
# Parent  ef8df0a8b358f8acf575811c07ad39bd1653de03
Fix history widget selecting two lines with one click (bug #53346).

* history-dock-widget.cc: Remove header file <QDesktopWidget> from list.
* history-dock-widget.cc (history_dock_widget::construct): Drop the code that
sets the QListView m_history_list_view grid size which then uses default
sizing.

diff --git a/libgui/src/history-dock-widget.cc b/libgui/src/history-dock-widget.cc
--- a/libgui/src/history-dock-widget.cc
+++ b/libgui/src/history-dock-widget.cc
@@ -24,17 +24,16 @@ along with Octave; see the file COPYING.
 #  include "config.h"
 #endif
 
 #include <QApplication>
 #include <QClipboard>
 #include <QVBoxLayout>
 #include <QMenu>
 #include <QScrollBar>
-#include <QDesktopWidget>
 #include <QCompleter>
 #include <QLabel>
 
 #include "error.h"
 #include "resource-manager.h"
 
 #include "cmd-hist.h"
 
@@ -336,17 +335,11 @@ namespace octave
     connect (m_filter_checkbox, SIGNAL (toggled (bool)),
              this, SLOT (filter_activate (bool)));
     connect (m_filter->lineEdit (), SIGNAL (editingFinished (void)),
              this, SLOT (updatem_filter_history (void)));
 
     connect (m_history_list_view, SIGNAL (doubleClicked (QModelIndex)),
              this, SLOT (handle_double_click (QModelIndex)));
 
-    // shrink max. displayed entry size to desktop width
-    QSize screen = QDesktopWidget ().screenGeometry ().size ();
-    int w = screen.width ();
-    QFontMetrics fm = m_history_list_view->fontMetrics ();
-    int h = fm.height ();
-    m_history_list_view->setGridSize (QSize (w,h));
     m_history_list_view->setTextElideMode (Qt::ElideRight);
   }
 }
